src/amo/components/VPNPromoBanner/index.js

```diff
@@ -69,28 +90,37 @@ export class VPNPromoBannerBase extends React.Component<InternalProps> {
     );
   }
 
-  onButtonClick: () => void = () => {
-    this.props._tracking.sendEvent({
-      action: VPN_PROMO_CLICK_ACTION,
+  onInteract: (action: string) => void = (action) => {
+    const { _tracking, _window } = this.props;
+
+    const impressionCount = getImpressionCount(_window);
+    _tracking.sendEvent({
+      action,
       category: VPN_PROMO_CATEGORY,
+      label: String(impressionCount),
```

Comment:
The issue says:

> we can include the number of impressions that proceeded it. We could send this in the **value field** of the event, or we could use a custom metric, which might make reporting a bit easier/friendlier.

Reply:
I should update the issue. Upon experimentation, I found that the best way to enable reporting in the manner we want is to simply make this a label. That way it can be treated as a dimension. We don't really need a custom metric or a value as we're not particularly interested in aggregating this data.

Reply:
> as we're not particularly interested in aggregating this data.

mm why not? Aren't we interested in knowning how many users bought the VPN after _N_ impressions?

Reply:
We can see that with this approach. Making impression count a label enables it as a dimension, so we can easily generate a report that shows the number of users who clicked on the CTA after *N* impressions. A custom metric would allow us to run calculations on that *N* number, for example, "what was the average number of impressions before a click". I'm not sure we care as much about that answer as we do about the breakdown by *N*, but I have asked that question in the issue.

Reply:
ah ok, I see (I think)

Reply:
If we decide to introduce a custom metric, we can do that in a follow up. I am awaiting a response on that.